test/intercept_test.dart

```diff
@@ -1,6 +1,11 @@
 import 'package:mobx/mobx.dart';
+import 'package:mobx/src/core/atom.dart';
+import 'package:mobx/src/interceptable.dart';
+import 'package:mockito/mockito.dart';
 import 'package:test/test.dart';
 
+import 'shared_mocks.dart';
+
 void main() {
   test('intercept', () {
     final x = observable(10);
```

Comment:
I think we also need a test where the `MockContext` is passed into the observable, action, reaction API directly. This will test if the context flows through properly and used correctly as well. I think the entire public API is missing the custom context tests?? 

This will be a good blackbox test of the context flowing from the user-level API

Reply:
MockContext tests exist for `Interceptors`, `action()`, `computed()`, `observable()` and `reaction()`

Reply:
Yup you are right. I think they are only missing for `autorun` and `when`

Reply:
Added context tests for `autorun` and `when`